CodePush.js

```diff
@@ -56,14 +64,19 @@ async function checkForUpdate(deploymentKey = null) {
    *    the currently running update. This should _never_ happen, unless there is a
    *    bug in the server, but we're adding this check just to double-check that the
    *    client app is resilient to a potential issue with the update check.
+   * 4) The server said there is an update, but the update's hash is the same as that
+   *    of the binary's currently running version. This should only happen in Android -
+   *    unlike iOS, we don't attach the binary's hash to the updateCheck request
+   *    because we want to avoid having to install diff updates against the binary's
+   *    version, which we can't do yet on Android.
    */
-  if (!update || update.updateAppVersion || localPackage && (update.packageHash === localPackage.packageHash)) {
+  if (!update || update.updateAppVersion || localPackage && (update.packageHash === localPackage.packageHash) || config.packageHash === update.packageHash) {
```

Comment:
For the last expression, don't you also want to check that `localPackage` is `null`? Since we wouldn't care about the binary's hash if the user was running on a CodePush update?